### PR TITLE
Connection url

### PR DESCRIPTION
**Pull request by rodneyhbrown7**
This PR is to resolve the issue that requires users to input   the URL of the form https://  when using the client.  This PR will look for https:// or http://  if neither is found it will prepend https:// to the provided hostname or IP address provided.   

This should also remove the requirement in ManageIQ to enter https://<ip> in the hostname field.

@miq_bot assign @walteraa

### CODE REVIEW

File: lib/xclarity_client/xclarity_base.rb [hunk 1]
```diff
@@ -16,7 +16,11 @@ def initialize(conf, uri)
     def connection_builder(conf, uri)
       $lxca_log.info "XClarityClient::XClarityBase connection_builder", "Creating connection to #{conf.host + uri}" 
       #Building configuration
-      @conn = Faraday.new(url: conf.host + uri) do |faraday|
+      host = conf.host;
```
Comment (skovic) [inline lib/xclarity_client/xclarity_base.rb]: Could probably do without the semicolon

File: spec/xclarity_client_node_spec.rb [hunk 2]
```diff
@@ -2,44 +2,45 @@
...
-    WebMock.allow_net_connect! # -- This line should be uncommented if you're using external mock test
+    # -- The next line should be uncommented
+    # if you're using external mock test
+    WebMock.allow_net_connect!
 
     conf = XClarityClient::Configuration.new(
       username:   ENV['LXCA_USERNAME'],
       password:   ENV['LXCA_PASSWORD'],
-      host:       ENV['LXCA_HOST'],
+      host:       'example.com',
```
Comment (skovic) [inline spec/xclarity_client_node_spec.rb]: Should this be changed back to ENV['LXCA_HOST'] or was this change intentional?

File: lib/xclarity_client/xclarity_base.rb (hunk 1 above)
Comment (rodneyhbrown7) [inline lib/xclarity_client/xclarity_base.rb]: That would probably be best. I will make that change.

File: spec/xclarity_client_node_spec.rb (hunk 2 above)
Comment (rodneyhbrown7) [inline spec/xclarity_client_node_spec.rb]: I changed it to test that 'https://' was prepended properly, but yes it should be changed back.  I will update that as well.

File: lib/xclarity_client/xclarity_base.rb [hunk 3]
```diff
@@ -16,7 +16,11 @@ def initialize(conf, uri)
...
+      if !host.start_with?('https://') && !host.start_with?('http://')
+         host = 'https://' + host
```
Comment (rodneyhbrown7) [inline lib/xclarity_client/xclarity_base.rb]: Yes. As far as I know.

File: lib/xclarity_client/xclarity_base.rb [hunk 4]
```diff
@@ -16,7 +16,11 @@ def initialize(conf, uri)
...
+      end
+      @conn = Faraday.new(url: host + uri) do |faraday|
```
Comment (rodneyhbrown7) [inline lib/xclarity_client/xclarity_base.rb]: I agree.

File: lib/xclarity_client/xclarity_base.rb (hunk 3 above)
Comment (walteraa) [inline lib/xclarity_client/xclarity_base.rb]: Ok, but is there a possibility to use the HTTPS procol over other port? If yes, it could be a problem

Comment (rodneyhbrown7) [inline lib/xclarity_client/xclarity_base.rb]: Even if they use HTTPS over another port I am not proposing to change the port information that was passed in. We are only adding https:// to the front of any URI that does not specify the protocol.  If they do specify a protocol & port, we will not modify the URI at all.

Comment (rodneyhbrown7) [inline lib/xclarity_client/xclarity_base.rb]: The other option is to push some of this logic up into the provider and have the provider add the protocol to be used if it is not specified.

Comment (walteraa) [inline lib/xclarity_client/xclarity_base.rb]: When I told about the port, it's other concern that we should analyze, maybe more problematic than if always will run over HTTPS. But we can discuss it in another moment.

Once the LXCA will always run over HTTPS protocol, as you said, it LGTM! 😄 👍

Comment (rodneyhbrown7) [inline lib/xclarity_client/xclarity_base.rb]: I just added a commit to refactor how this is implemented.   With the refactor if any protocol is specified we will use the specified protocol ( instead of just http and https).  If no protocol is specified we will assume https and use the port details provided.   I believe that MIQ is passing down port details when it constructus the provider connection.

**Comment by walteraa** [comment]
Waiting for @skovic approval to merge it
